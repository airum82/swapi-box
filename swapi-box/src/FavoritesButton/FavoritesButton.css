.favorites-count {
  background-color: #ce854e;
  margin: 0;
  padding-top: 56%; }

.favorite-button {
  padding-right: 33%;
  padding-left: 16%; }
