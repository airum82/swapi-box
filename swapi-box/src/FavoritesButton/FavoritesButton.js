import React from 'react';

export const FavoritesButton = (props) => {
  return (
    <button>favorites</button>

  )
}