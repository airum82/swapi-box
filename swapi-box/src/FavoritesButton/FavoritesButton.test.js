import React from 'react';
import { shallow, mount } from 'enzyme';
import { FavoritesButton } from './FavoritesButton';

describe('FavoritesButton', () => {
  it('Should match snapshot upon render', () => {
    const wrapper = shallow(<FavoritesButton
                            />);

    expect(wrapper).toMatchSnapshot();
  })

  it('Should invoke the viewFavorites function on click', () => {
    const wrapper = shallow(<FavoritesButton
                            />);

    wrapper.instance().viewFavorites = jest.fn();

    wrapper.find('.favorite-button').simulate('click');
    expect(wrapper.instance().viewFavorites).toHaveBeenCalled();
  })
})