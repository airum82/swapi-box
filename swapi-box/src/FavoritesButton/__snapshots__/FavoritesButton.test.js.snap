// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoritesButton Should match snapshot upon render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoritesButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="favorite-button"
          onClick={[Function]}
>
          favorites
</button>,
        <span
          className="favorites-count"
>
           
          0
</span>,
      ],
      "className": "favorites",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "favorites",
          "className": "favorite-button",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "favorites",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            0,
          ],
          "className": "favorites-count",
        },
        "ref": null,
        "rendered": Array [
          " ",
          0,
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="favorite-button"
            onClick={[Function]}
>
            favorites
</button>,
          <span
            className="favorites-count"
>
             
            0
</span>,
        ],
        "className": "favorites",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "favorites",
            "className": "favorite-button",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "favorites",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              0,
            ],
            "className": "favorites-count",
          },
          "ref": null,
          "rendered": Array [
            " ",
            0,
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
